lib/engine.py

```diff
@@ -541,11 +540,12 @@ def format_insert_value(self, value, datatype):
         elif datatype == "char":
             if strvalue.lower() in nulls:
                 return "null"
-            # automatically escape quotes in string fields
-            if hasattr(self.table, "escape_double_quotes") and self.table.escape_double_quotes:
-                strvalue = self.escape_double_quotes(strvalue)
-            if hasattr(self.table, "escape_single_quotes") and self.table.escape_single_quotes:
-                strvalue = self.escape_single_quotes(strvalue)
+            if escape:
+                # automatically escape quotes in string fields
+                if hasattr(self.table, "escape_double_quotes") and self.table.escape_double_quotes:
+                    strvalue = self.escape_double_quotes(strvalue)
+                if hasattr(self.table, "escape_single_quotes") and self.table.escape_single_quotes:
+                    strvalue = self.escape_single_quotes(strvalue)
```

Comment:
Ah, very interesting. Really nice solution.

After reading through the docstring and the function I think a broader rewording is really needed to capture what's going on with this function anyway. I gave it a go and came up with:

```
        """Format a value for an insert statement based on data type

        Different data types need to be formated differently to be properly
        stored in database management systems. The correct formats are
        obtained by:

        1. Removing extra enclosing quotes
        2. Harmonizing null indicators
        3. Cleaning up badly formatted integers
        4. Obtaining consistent float representations of decimals

        The optional `escape` argument controls whether additional quotes in
        strings are escaped, as needed for SQL database management systems
        (escape=True), or not escaped, as needed for flat file based engines
        (escape=False). """
```